core/src/main/java/nl/nn/adapterframework/receivers/JavaListener.java

```diff
@@ -113,19 +113,30 @@ public synchronized void close() throws ListenerException {
 			unregisterListener();
 			// unregister from global list
 			if (StringUtils.isNotEmpty(getServiceName())) {
-				// Current DispatcherManager (version 1.3) doesn't have an
-				// unregister method, instead a call to register with a null
-				// value is done.
-				DispatcherManagerFactory.getDispatcherManager().register(getServiceName(), null);
+				DispatcherManagerFactory.getDispatcherManager().unregister(getServiceName());
 			}
 		}
 		catch (Exception e) {
-			throw new ListenerException("error occured while stopping listener [" + getName() + "]", e);
+			throw new ListenerException("error occurred while stopping listener [" + getName() + "]", e);
 		}
 	}
 
 	@Override
 	public String processRequest(String correlationId, String rawMessage, HashMap context) throws ListenerException {
+		try {
+			Message result = processRequest(correlationId, (M) rawMessage, new Message(rawMessage), context);
+			return result.asString();
+		} catch (IOException e) {
+			throw new ListenerException("cannot convert stream", e);
+		}
+	}
+
+	@Override
+	public Message processRequest(String correlationId, Message message, PipeLineSession session) throws ListenerException {
+		return processRequest(correlationId, (M) message.asObject(), message, session);
```

Comment:
Ik denk dat het verstandig is de response hier te registreren als closeOnClose op de nieuwe session.
```suggestion
Message response = processRequest(correlationId, (M) message.asObject(), message, session);
session.closeOnClose(response);
return  response;
```